src/glob.rs

```diff
@@ -234,20 +252,20 @@ pub fn read_globs_from_dir<P: AsRef<Path>>(dir: P) -> Vec<Glob> {
 }
 
 pub struct GlobMap {
-    globs: Vec<Glob>,
+    globs: HashSet<Glob>,
 }
 
 impl GlobMap {
     pub fn new() -> GlobMap {
-        GlobMap { globs: Vec::new() }
+        GlobMap { globs: HashSet::new() }
     }
 
     pub fn add_glob(&mut self, glob: Glob) {
-        self.globs.push(glob);
+        self.globs.replace(glob);
```

Comment:
This should be `insert(glob)`, if we want to maintain the same behaviour as xdgmime.

Reply:
Oops yes my bad, that's what I meant but I copied the wrong line from the previous code.